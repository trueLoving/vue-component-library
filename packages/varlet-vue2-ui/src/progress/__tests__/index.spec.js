import example from '../example'
import Vue from 'vue'
import Progress from '..'
import VarProgress from '../Progress'
import { mount } from '@vue/test-utils'

test('test progress example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
})

test('test progress plugin', () => {
  Vue.use(Progress)
  expect(Vue.component(Progress.name)).toBeTruthy()
})

test('test linear progress prop', () => {
  const wrapper = mount(VarProgress, {
    propsData: {
      value: 30,
      lineWidth: 8,
      color: '#ff9800',
      trackColor: '#f5cb90',
      ripple: true,
      label: true,
    },
    scopedSlots: {
      default: '<div>success</div>',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test circle progress prop', () => {
  const wrapper = mount(VarProgress, {
    propsData: {
      mode: 'circle',
      value: 30,
      lineWidth: 8,
      color: '#ff9800',
      trackColor: '#f5cb90',
      size: 50,
      rotate: 10,
      label: true,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})
