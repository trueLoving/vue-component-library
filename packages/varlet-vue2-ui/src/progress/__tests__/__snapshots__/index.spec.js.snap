// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test circle progress prop 1`] = `
"<div class=\\"var-progress\\">
  <!---->
  <div class=\\"var-progress-circle\\" style=\\"width: 50px; height: 50px;\\"><svg viewBox=\\"0 0 50 50\\" class=\\"var-progress-circle__svg\\" style=\\"transform: rotate(-80deg);\\">
      <circle cx=\\"25\\" cy=\\"25\\" r=\\"21\\" fill=\\"transparent\\" stroke-width=\\"8\\" class=\\"var-progress-circle__background\\" style=\\"stroke-dasharray: 131.94689145077132; stroke: #f5cb90;\\"></circle>
      <circle cx=\\"25\\" cy=\\"25\\" r=\\"21\\" fill=\\"transparent\\" stroke-width=\\"8\\" class=\\"var-progress-circle__certain\\" style=\\"stroke-dasharray: 39.5840674352314, 131.94689145077132; stroke: #ff9800;\\"></circle>
    </svg>
    <div class=\\"var-progress-circle__label\\">
      30%
    </div>
  </div>
</div>"
`;

exports[`test linear progress prop 1`] = `
"<div class=\\"var-progress\\">
  <div class=\\"var-progress-linear\\">
    <div class=\\"var-progress-linear__block\\" style=\\"height: 8px;\\">
      <div class=\\"var-progress-linear__background\\" style=\\"background: rgb(245, 203, 144);\\"></div>
      <div class=\\"var-progress-linear__certain var-progress-linear__ripple\\" style=\\"background: rgb(255, 152, 0); width: 30%;\\"></div>
    </div>
    <div class=\\"var-progress-linear__label\\">
      <div>success</div>
    </div>
  </div>
  <!---->
</div>"
`;

exports[`test progress example 1`] = `
"<div class=\\"var-progress-example\\">
  <div class=\\"space\\"></div>
  <div class=\\"progress-demo__basic\\">
    <div class=\\"app-type\\">基本使用</div>
    <div class=\\"var-progress\\">
      <div class=\\"var-progress-linear\\">
        <div class=\\"var-progress-linear__block\\" style=\\"height: 4px;\\">
          <!---->
          <div class=\\"var-progress-linear__certain\\" style=\\"width: 20%;\\"></div>
        </div>
        <!---->
      </div>
      <!---->
    </div>
    <div class=\\"var-progress\\">
      <div class=\\"var-progress-linear\\">
        <div class=\\"var-progress-linear__block\\" style=\\"height: 4px;\\">
          <!---->
          <div class=\\"var-progress-linear__certain\\" style=\\"width: 0%;\\"></div>
        </div>
        <!---->
      </div>
      <!---->
    </div>
    <div class=\\"var-progress\\">
      <div class=\\"var-progress-linear\\">
        <div class=\\"var-progress-linear__block\\" style=\\"height: 4px;\\">
          <!---->
          <div class=\\"var-progress-linear__certain\\" style=\\"width: 100%;\\"></div>
        </div>
        <!---->
      </div>
      <!---->
    </div>
  </div>
  <div class=\\"progress-demo__label\\">
    <div class=\\"app-type\\">显示标签</div>
    <div class=\\"var-progress\\">
      <div class=\\"var-progress-linear\\">
        <div class=\\"var-progress-linear__block\\" style=\\"height: 4px;\\">
          <div class=\\"var-progress-linear__background\\"></div>
          <div class=\\"var-progress-linear__certain\\" style=\\"width: 30%;\\"></div>
        </div>
        <div class=\\"var-progress-linear__label\\">
          30%
        </div>
      </div>
      <!---->
    </div>
    <div class=\\"var-progress\\">
      <div class=\\"var-progress-linear\\">
        <div class=\\"var-progress-linear__block\\" style=\\"height: 4px;\\">
          <div class=\\"var-progress-linear__background\\"></div>
          <div class=\\"var-progress-linear__certain\\" style=\\"width: 0%;\\"></div>
        </div>
        <div class=\\"var-progress-linear__label\\">
          0%
        </div>
      </div>
      <!---->
    </div>
    <div class=\\"var-progress\\">
      <div class=\\"var-progress-linear\\">
        <div class=\\"var-progress-linear__block\\" style=\\"height: 4px;\\">
          <div class=\\"var-progress-linear__background\\"></div>
          <div class=\\"var-progress-linear__certain\\" style=\\"width: 100%;\\"></div>
        </div>
        <div class=\\"var-progress-linear__label\\">success</div>
      </div>
      <!---->
    </div>
  </div>
  <div class=\\"progress-demo__style\\">
    <div class=\\"app-type\\">自定义样式</div>
    <div class=\\"var-progress\\">
      <div class=\\"var-progress-linear\\">
        <div class=\\"var-progress-linear__block\\" style=\\"height: 8px;\\">
          <div class=\\"var-progress-linear__background\\"></div>
          <div class=\\"var-progress-linear__certain\\" style=\\"background: rgb(255, 159, 0); width: 30%;\\"></div>
        </div>
        <!---->
      </div>
      <!---->
    </div>
    <div class=\\"var-progress\\">
      <div class=\\"var-progress-linear\\">
        <div class=\\"var-progress-linear__block\\" style=\\"height: 8px;\\">
          <div class=\\"var-progress-linear__background\\" style=\\"background: rgb(245, 203, 144);\\"></div>
          <div class=\\"var-progress-linear__certain\\" style=\\"background: rgb(255, 159, 0); width: 60%;\\"></div>
        </div>
        <!---->
      </div>
      <!---->
    </div>
    <div class=\\"var-progress\\">
      <div class=\\"var-progress-linear\\">
        <div class=\\"var-progress-linear__block\\" style=\\"height: 8px;\\">
          <div class=\\"var-progress-linear__background\\" style=\\"background: rgb(245, 203, 144);\\"></div>
          <div class=\\"var-progress-linear__certain var-progress-linear__ripple\\" style=\\"background: rgb(255, 159, 0); width: 80%;\\"></div>
        </div>
        <!---->
      </div>
      <!---->
    </div>
  </div>
  <div class=\\"app-type\\">环形进度条</div>
  <div class=\\"progress-demo__circle\\">
    <div class=\\"var-progress\\">
      <!---->
      <div class=\\"var-progress-circle\\" style=\\"width: 56px; height: 56px;\\"><svg viewBox=\\"0 0 56 56\\" class=\\"var-progress-circle__svg\\" style=\\"transform: rotate(-90deg);\\">
          <circle cx=\\"28\\" cy=\\"28\\" r=\\"25.5\\" fill=\\"transparent\\" stroke-width=\\"5\\" class=\\"var-progress-circle__background\\" style=\\"stroke-dasharray: 160.22122533307945;\\"></circle>
          <circle cx=\\"28\\" cy=\\"28\\" r=\\"25.5\\" fill=\\"transparent\\" stroke-width=\\"5\\" class=\\"var-progress-circle__certain\\" style=\\"stroke-dasharray: 48.06636759992383, 160.22122533307945;\\"></circle>
        </svg>
        <!---->
      </div>
    </div>
    <div class=\\"space\\"></div>
    <div class=\\"var-progress\\">
      <!---->
      <div class=\\"var-progress-circle\\" style=\\"width: 56px; height: 56px;\\"><svg viewBox=\\"0 0 56 56\\" class=\\"var-progress-circle__svg\\" style=\\"transform: rotate(-90deg);\\">
          <circle cx=\\"28\\" cy=\\"28\\" r=\\"25.5\\" fill=\\"transparent\\" stroke-width=\\"5\\" class=\\"var-progress-circle__background\\" style=\\"stroke-dasharray: 160.22122533307945;\\"></circle>
          <circle cx=\\"28\\" cy=\\"28\\" r=\\"25.5\\" fill=\\"transparent\\" stroke-width=\\"5\\" class=\\"var-progress-circle__certain\\" style=\\"stroke-dasharray: 0, 160.22122533307945;\\"></circle>
        </svg>
        <div class=\\"var-progress-circle__label\\">
          0%
        </div>
      </div>
    </div>
    <div class=\\"space\\"></div>
    <div class=\\"var-progress\\">
      <!---->
      <div class=\\"var-progress-circle\\" style=\\"width: 56px; height: 56px;\\"><svg viewBox=\\"0 0 56 56\\" class=\\"var-progress-circle__svg\\" style=\\"transform: rotate(-90deg);\\">
          <circle cx=\\"28\\" cy=\\"28\\" r=\\"25.5\\" fill=\\"transparent\\" stroke-width=\\"5\\" class=\\"var-progress-circle__background\\" style=\\"stroke-dasharray: 160.22122533307945;\\"></circle>
          <circle cx=\\"28\\" cy=\\"28\\" r=\\"25.5\\" fill=\\"transparent\\" stroke-width=\\"5\\" class=\\"var-progress-circle__certain\\" style=\\"stroke-dasharray: 160.22122533307945, 160.22122533307945;\\"></circle>
        </svg>
        <div class=\\"var-progress-circle__label\\">
          100%
        </div>
      </div>
    </div>
  </div>
  <div class=\\"app-type\\">隐藏轨道</div>
  <div class=\\"progress-demo__circle\\">
    <div class=\\"var-progress\\">
      <!---->
      <div class=\\"var-progress-circle\\" style=\\"width: 56px; height: 56px;\\"><svg viewBox=\\"0 0 56 56\\" class=\\"var-progress-circle__svg\\" style=\\"transform: rotate(-90deg);\\">
          <!---->
          <circle cx=\\"28\\" cy=\\"28\\" r=\\"26\\" fill=\\"transparent\\" stroke-width=\\"4\\" class=\\"var-progress-circle__certain\\" style=\\"stroke-dasharray: 81.68140899333463, 163.36281798666926;\\"></circle>
        </svg>
        <!---->
      </div>
    </div>
  </div>
</div>"
`;
